rasa/nlu/featurizers/dense_featurizer/spacy_featurizer.py

```diff
@@ -51,6 +62,20 @@ def process(self, message: Message, **kwargs: Any) -> None:
 
         self._set_spacy_features(message)
 
+    def _calculate_cls_vector(self, features: np.ndarray) -> np.ndarray:
+        # take only non zeros feature vectors into account
+        features = np.array([f for f in features if f.any()])
```

Comment:
```suggestion
        non_zero_features = np.array([f for f in features if f.any()])
```
to make it a bit more noticeable and explicit